### PR TITLE
Adds validation to report enrollments

### PR DESCRIPTION
**Pull request by jlhogan**

### CODE REVIEW

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/UtilizationTable.tsx [hunk 1]
```diff
@@ -37,39 +38,41 @@ export default function UtilizationTable(report: CdcReport) {
...
+      }
+    })
+    : [];
 
-    return ageGroups.flatMap(ageGroup => fundingTimes.flatMap(fundingTime => {
+  const rows: UtilizationTableRow[] = ageGroups.flatMap(ageGroup => 
+    fundingTimes.flatMap(fundingTime => {
       const count = enrollments
-        .filter(enrollment => enrollment.ageGroup === ageGroup && enrollment.time === fundingTime)
+        .filter(enrollment => enrollment.ageGroup == ageGroup && enrollment.time == fundingTime)
```
Comment (cmgiven) [inline src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/UtilizationTable.tsx]: Why this change?

File: src/Hedwig/Repositories/ReportRepository.cs [hunk 2]
```diff
@@ -68,43 +68,17 @@ public async Task<CdcReport> GetReportForOrganizationAsync(int id, int orgId, st
       var reportResult = await reportQuery.FirstOrDefaultAsync();
 
       // Optionally manually insert time-versioned enrollment records
-      if (include.Contains(INCLUDE_ENROLLMENTS) || include.Contains(INCLUDE_CHILD))
+      if (include.Contains(INCLUDE_ENROLLMENTS))
```
Comment (cmgiven) [inline src/Hedwig/Repositories/ReportRepository.cs]: This change means that `INCLUDE_CHILD` is a no-op unless `INCLUDE_ENROLLMENTS` is also included, yes? Is there a reason to prefer this?

File: src/Hedwig/Controllers/ReportsController.cs [hunk 3]
```diff
@@ -16,10 +17,15 @@ namespace Hedwig.Controllers
   [Route("api/organizations/{orgId:int}/[controller]")]
   public class ReportsController : ControllerBase
   {
+    private readonly INonBlockingValidator _validator;
```
Comment (cmgiven) [inline src/Hedwig/Controllers/ReportsController.cs]: I'd expect we'd also want to validate on PUT to prevent submitting a report that has issues?

File: src/Hedwig/Repositories/ReportRepository.cs (hunk 2 above)
Comment (jlhogan) [reply src/Hedwig/Repositories/ReportRepository.cs]: that's the way we implemented all the enrollment includes, but it's not necessary. fine w/ either.

File: src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/UtilizationTable.tsx (hunk 1 above)
Comment (jlhogan) [reply src/Hedwig/ClientApp/src/containers/Reports/ReportDetail/UtilizationTable.tsx]: yeah good q.... :)
